src/Presentation.ConsoleApplication/Program.cs

```diff
@@ -0,0 +1,122 @@
+            {
+                Program.Run(parentServiceProvider);
+            }
+        }
+
+        private static IStringKeyValueStore GetConfigurationKeyValueStore()
+        {
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddPetProjectConsulServices(Program.Configuration, true);
+            serviceCollection.AddSingleton<ILogger>(NullLogger.Instance);
+
+            using (var tempProvider = serviceCollection.BuildServiceProvider())
+            {
+                return tempProvider.GetRequiredService<IStringKeyValueStore>();
+            }
+        }
+
+        private static void SetupConfiguration()
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                .AddEnvironmentVariables("MTS_APP_SETTINGS_");
+
+            Program.Configuration = builder.Build();
+        }
+
+        private static void SetupServices(IServiceCollection serviceCollection, IStringKeyValueStore configStore)
+        {
+            Program.SetupLogging(serviceCollection, configStore);
+
+            serviceCollection.AddPetProjectConsulServices(Program.Configuration, true);
+
+            serviceCollection.SetupDependencies();
+        }
+
+        private static void SetupLogging(IServiceCollection serviceCollection, IStringKeyValueStore configStore)
+        {
+            var kafkaConfig = new Framework.Logging.Producer.KafkaConfiguration
+            {
+                Brokers = configStore.GetAndConvertValue<string>("Logging/KafkaConfiguration/Brokers").Split(','),
+                Topic = configStore.GetAndConvertValue<string>("Logging/KafkaConfiguration/Topic")
+            };
+
+            var sinkConfig = new PeriodicSinkConfiguration
+            {
+                BatchSizeLimit = configStore.GetAndConvertValue<int>("Logging/BatchSizeLimit"),
+                Period = TimeSpan.FromMilliseconds(configStore.GetAndConvertValue<int>("Logging/PeriodMs"))
+            };
+
+            var logLevel = configStore.GetAndConvertValue<LogEventLevel>("Logging/LogLevel");
+            var logType = configStore.GetAndConvertValue<string>("Logging/LogType");
+
+            serviceCollection.AddLogging(builder => builder.AddPetProjectLogging(logLevel, sinkConfig, kafkaConfig, logType, true));
+        }
+
+        private static void Run(IServiceProvider scopedProvider)
+        {
+            var logger = scopedProvider.GetService<ILoggerFactory>().CreateLogger<Program>();
+
+            logger.LogCritical("Starting MicroTransactionsUpdater...");
+
+            using (var consumer = scopedProvider.GetRequiredService<IConsumer<TransactionEvent>>())
+            {
+                var task = Task.Factory.StartNew(() => consumer.StartConsuming(), TaskCreationOptions.LongRunning);
```

Comment:
This isn't needed anymore, the StartConsuming method already creates a LongRunning task.

You only need to do consumer.StarConsuming

Reply:
done